crates/types/src/services/block_producer.rs

```diff
@@ -23,6 +23,9 @@ pub struct Components<Source> {
     pub coinbase_recipient: ContractId,
     /// The gas price for all transactions in the block.
     pub gas_price: u64,
+    /// Chain ID
+    #[cfg(feature = "fault-proving")]
+    pub chain_id: crate::fuel_types::ChainId,
```

Comment:
> producer/executor codeowners: The chainId has to be bubbled all the way from the producer until the generation of a block header from a partial block header because the tx_id_commitment needs it

This doesn't seem to be the most logical place for this value. I think of components as being things unique to the specific block, whereas the `chain_id` shouldn't change during a session.

The executor already has the `chain_id` via the `ConsensusParameters`. Could we get it from the executor when we convert the `Components` into the block?:
<img width="758" alt="image" src="https://github.com/user-attachments/assets/6646809f-a5a6-4e2b-8cca-28cc0f285a83" />

Reply:
The `block_executor` should also have access to it.